src/ngircd/login.c

```diff
@@ -201,9 +201,40 @@ Login_User_PostAuth(CLIENT *Client)
 	} else
 		IRC_SetPenalty(Client, 1);
 
+  /* Autojoin clients to the channels */
+  Login_Autojoin(Client);
+
 	return CONNECTED;
 }
 
+/**
+ * Autojoin clients to the channels set by administrator
+ * If autojoin is not set in Config or the channel is not available for search - do nothing
+ *
+ **/
+GLOBAL void 
+Login_Autojoin(CLIENT *Client)
+{
+	/** make an autojoin to each channel that is good for it **/
+	const struct Conf_Channel *conf_chan;
+	size_t i, n, channel_count = array_length(&Conf_Channels, sizeof(*conf_chan));
+	conf_chan = array_start(&Conf_Channels);
+	assert(channel_count == 0 || conf_chan != NULL);
+	REQUEST Req;
```

Comment:
This line fails with OpenWatcom 1.9:
  ERROR! E1058: Cannot use typedef 'REQUEST' as a variable

Reply:
Apparently the declaration needs to be above anything else; moving it to line 218 solved the problem. The patch now applies mostly-cleanly to ngIRCd-DOS, compiles with Open Watcom 1.9, and runs on my 286. :)